src/services/pool.ts

```diff
@@ -109,9 +143,42 @@ const mapBranches = (branches: BranchEntry[], fullBaseUrl: string) => {
   }));
 };
 
-const mapRepos = (repo: RepoDocument): RepoResponse => ({
-  repoName: repo.repoName,
-  project: repo.project,
-  search: repo.search,
-  branches: mapBranches(repo.branches, getRepoUrl(repo.url[ENV_URL_KEY], repo.prefix[ENV_URL_KEY])),
-});
+const mapRepos = async (repo: RepoDocument) => {
+  try {
+    const query: Filter<DocsetDocument> = { project: repo.project };
+    const strictOptions: FindOptions = {
+      projection: {
+        project: 1,
+        url: 1,
+        prefix: 1,
+      },
+    };
+
+    const repo_docset = await db
+      .collection<DocsetDocument>(DOCSETS_COLLECTION)
+      .find(query, strictOptions)
+      .map(mapDocsets)
+      .toArray();
+
+    return {
+      repoName: repo.repoName,
+      project: repo.project,
+      search: repo.search,
+      branches: mapBranches(
+        repo.branches,
+        getRepoUrl(repo_docset[0].url[ENV_URL_KEY], repo_docset[0].prefix[ENV_URL_KEY])
```

Comment:
This might not be possible given our current data, but I think it might be a good idea to handle a case where 0 results are found for a docset to prevent any future hiccups.

Maybe we can just return an empty branches array?